staging/src/k8s.io/component-base/featuregate/feature_gate.go

```diff
@@ -296,6 +307,32 @@ func (f *featureGate) Add(features map[Feature]FeatureSpec) error {
 	return nil
 }
 
+func (f *featureGate) OverrideDefault(name Feature, override bool) error {
+	f.lock.Lock()
+	defer f.lock.Unlock()
+
+	if f.closed {
+		return fmt.Errorf("cannot override feature gate defaults after adding it to the flag set")
+	}
+
+	known := f.known.Load().(map[Feature]FeatureSpec)
+
+	if spec, ok := known[name]; !ok {
+		return fmt.Errorf("cannot override default: feature %q is not registered", name)
+	} else if spec.LockToDefault && override != spec.Default {
+		return fmt.Errorf("cannot override default: feature %q default is locked to %t", name, spec.Default)
+	}
+
+	defaultOverrides := map[Feature]bool{}
```

Comment:
can't just directly modify the map?  modifying the map will keep all consuming code identical

Reply:
It's stored separately here to keep the existing behavior on duplicate registrations since the order of calls to Add is likely to depend on package initialization order and be nonobvious:

```go
fg.Add(map[Feature]FeatureSpec{"Foo": {Default: false}})
fg.OverrideDefault(map[Feature]bool{"Foo": true})
fg.Add(map[Feature]FeatureSpec{"Foo": {Default: true}}) // nil error?
```

But I'm happy to simplify this and make the OverrideDefault doc clear on safe usage.